### PR TITLE
Pcb rules

### PR DESCRIPTION
**Pull request by akmorrison**

### CODE REVIEW

File: standard.tex [hunk 1]
```diff
@@ -66,15 +69,41 @@ \subsection{General Wiring \& Assembly}
...
 \item All software should use a VCS (probably git), and all final versions of your software should go with your development logs (see Documentation Standards) on the drive.
 \end{itemize}
+%end of software subsection -- }}}
 
+%start of pcb subsection -- {{{
+\subsection{PCB design}
+\begin{itemize}
+\item Watch this EEVblog video: \url{https://www.youtube.com/watch?v=JrH_itjMDjo}. It's like an hour and a half long, but well worth watching.
+\item Given the choice, please draft your schematics and board layouts in KiCad. It's free (GPLv3), used by the rest of the team, and runs on all platforms. \url{http://kicad-pcb.org/}
+\item if your PCBs are meant to be screwed onto something, put those mounting holes on your layout, and if at all possible, use M3 screws (for standardization purposes, if you have a space restriction, smaller screws are fine).
```
Comment (amihaila) [inline standard.tex]: "If your PCBs are meant to be screwed onto something, put mounting holes on your layout, and if at all possible, use M3 screws (for standardization purposes. If you have a space restriction, smaller screws are fine)."

File: standard.tex [hunk 2]
```diff
@@ -66,15 +69,41 @@ \subsection{General Wiring \& Assembly}
...
 \begin{itemize}
 \item If your system has any software anywhere, that software should be unit tested.
 \item If your system requires a computer program to be used, that program counts as mission critical. It should run on at least two machines and at least two people should be trained to use it.
 \item All mission critical software should be expressible (and accompanied by) less than 100 lines of pseudocode.
 \item All software should use a VCS (probably git), and all final versions of your software should go with your development logs (see Documentation Standards) on the drive.
 \end{itemize}
+%end of software subsection -- }}}
 
+%start of pcb subsection -- {{{
+\subsection{PCB design}
```
Comment (amihaila) [inline standard.tex]: *Design

File: standard.tex [hunk 3]
```diff
@@ -66,15 +69,41 @@ \subsection{General Wiring \& Assembly}
...
+\begin{itemize}
+\item Watch this EEVblog video: \url{https://www.youtube.com/watch?v=JrH_itjMDjo}. It's like an hour and a half long, but well worth watching.
+\item Given the choice, please draft your schematics and board layouts in KiCad. It's free (GPLv3), used by the rest of the team, and runs on all platforms. \url{http://kicad-pcb.org/}
+\item if your PCBs are meant to be screwed onto something, put those mounting holes on your layout, and if at all possible, use M3 screws (for standardization purposes, if you have a space restriction, smaller screws are fine).
+\item Before ordering any components, make a list of all components in your design, and go through their data sheets one by one and ensure that they meet or exceed the requirements of your design. Do not just trust the Digikey table values, they are occasionally mislabeled.
```
Comment (amihaila) [inline standard.tex]: "Do not just trust the Digikey table values. They are occasionally mislabeled."

File: standard.tex [hunk 4]
```diff
@@ -66,15 +69,41 @@ \subsection{General Wiring \& Assembly}
...
+\item The cheapest board that can be purchased in small quantities is a 2 sided 1oz 5cm by 10cm 10/10 (10 mil (.254mm) min clearance and min trace width) 0.4mm smallest drill size board. If at all possible, try to make your design fit that requirement.
+\item If a pin is listed as NC on a data sheet, don't pull it to ground. Just leave it unconnected. Unused IO pins are a different story, they may need to be shorted to ground or VCC. Consult your data sheet in this instance.
```
Comment (amihaila) [inline standard.tex]: c o m m a s p l i c e

File: standard.tex [hunk 5]
```diff
@@ -66,15 +69,41 @@ \subsection{General Wiring \& Assembly}
...
+
+\item If you plan to hand-solder your boards for assembly:
+\begin{itemize}
+\item Watch this video \url{https://www.youtube.com/watch?v=J5Sb21qbpEQ} (EEVblog soldering tutorial). Then watch part 2. Then watch part 3. Yes that involves an hour and a half time commitment, but after watching those 3 you'll know exactly as much as I do about soldering.
```
Comment (amihaila) [inline standard.tex]: "Watch this video \url{https://www.youtube.com/watch?v=J5Sb21qbpEQ} (EEVblog soldering tutorial) and its sequels."

The last sentence in that item doesn't add anything useful.

File: standard.tex [hunk 6]
```diff
@@ -66,15 +69,41 @@ \subsection{General Wiring \& Assembly}
...
+\item Use pads that are longer than you need. KiCad has specific footprints whose names end in \_handsoldering, which have extended pads for easier hand-soldering.
+\item Use flux. It makes surface mount soldering orders of magnitude cheaper. 10ml syringes are less than \$20 at Sayal, and should last for a while.
+\item SOT-23 and 0805 SMT components are quite easily hand-solderable. SOT-23-5 are also doable, but are harder. I've never attempted anything smaller than those.
```
Comment (amihaila) [inline standard.tex]: Last sentence doesn't add much.

File: standard.tex [hunk 7]
```diff
@@ -66,15 +69,41 @@ \subsection{General Wiring \& Assembly}
...
 \item If your system requires a computer program to be used, that program counts as mission critical. It should run on at least two machines and at least two people should be trained to use it.
 \item All mission critical software should be expressible (and accompanied by) less than 100 lines of pseudocode.
 \item All software should use a VCS (probably git), and all final versions of your software should go with your development logs (see Documentation Standards) on the drive.
 \end{itemize}
+%end of software subsection -- }}}
 
+%start of pcb subsection -- {{{
+\subsection{PCB design}
+\begin{itemize}
+\item Watch this EEVblog video: \url{https://www.youtube.com/watch?v=JrH_itjMDjo}. It's like an hour and a half long, but well worth watching.
```
Comment (amihaila) [inline standard.tex]: Comment on what's in this video would be useful.

File: standard.tex (hunk 3 above)
Comment (amihaila) [reply standard.tex]: Also, this isn't really a PCB design specific thing. It would fit better in another section.

File: standard.tex [hunk 8]
```diff
@@ -66,15 +69,41 @@ \subsection{General Wiring \& Assembly}
...
+\item Given the choice, please draft your schematics and board layouts in KiCad. It's free (GPLv3), used by the rest of the team, and runs on all platforms. \url{http://kicad-pcb.org/}
+\item if your PCBs are meant to be screwed onto something, put those mounting holes on your layout, and if at all possible, use M3 screws (for standardization purposes, if you have a space restriction, smaller screws are fine).
+\item Before ordering any components, make a list of all components in your design, and go through their data sheets one by one and ensure that they meet or exceed the requirements of your design. Do not just trust the Digikey table values, they are occasionally mislabeled.
+\item The cheapest board that can be purchased in small quantities is a 2 sided 1oz 5cm by 10cm 10/10 (10 mil (.254mm) min clearance and min trace width) 0.4mm smallest drill size board. If at all possible, try to make your design fit that requirement.
```
Comment (amihaila) [inline standard.tex]: What's the difference in cost? Is it actually significant or can this point be generally ignored?

File: standard.tex (hunk 4 above)
Comment (amihaila) [reply standard.tex]: Comment on why would also be good.

**Review by amihaila — COMMENTED** [review]
I mostly edited for form/clarity. I'll submit my own request for extra additions.

Also, was this indentation really necessary? I don't think it really helps readability in this case and it also breaks the git blame.

File: standard.tex (hunk 1 above)
Comment (akmorrison) [reply standard.tex]: changed to your version

File: standard.tex (hunk 7 above)
Comment (akmorrison) [reply standard.tex]: Changed to "EEVBlog has a great (albeit long) introduction to PCB layout/design: \url{https://www.youtube.com/watch?v=JrH_itjMDjo}. It's like an hour and a half long, but well worth watching."

File: standard.tex (hunk 2 above)
Comment (akmorrison) [reply standard.tex]: I capitalized it

File: standard.tex (hunk 3 above)
Comment (akmorrison) [reply standard.tex]: Agree with it not really fitting here, do you think there's enough advice to fill a whole pre-purchasing checklist? Like as a separate subsection to the rules section. For now I'll move it into "miscellaneous", but might be worth moving later

File: standard.tex (hunk 8 above)
Comment (akmorrison) [reply standard.tex]: I typically run everything through pcbshopper.com. At least in terms of drill size and copper weight, yes the price difference is noticeable, the 10/10 thing is less so.

File: standard.tex (hunk 4 above)
Comment (akmorrison) [reply standard.tex]: changed to "If a pin is listed as NC on a data sheet, don't pull it to ground. Just leave it unconnected. Some manufacturers (from what stack overflow tells me), use NC pins for in production testing, and as such they aren't actually high impedence pins, and shorting them elsewhere may destroy your IC."

File: standard.tex (hunk 5 above)
Comment (akmorrison) [reply standard.tex]: done

File: standard.tex (hunk 6 above)
Comment (akmorrison) [reply standard.tex]: It's there to say that "maybe it's possible to solder 0603 stuff by hand, but I don't know"

File: standard.tex (hunk 4 above)
Comment (amihaila) [reply standard.tex]: *impedance

Comment (akmorrison) [reply standard.tex]: True, I really need to turn on spell checking

**Comment by amihaila** [comment]
I've been reading through this again and some of the newer points don't really fit well into a "standard." This doc is supposed to be a set of general rules that we can reasonably check for and enforce and some of the newer stuff isn't that. Eg.
- what schematic editor you use. Doesn't matter as long as you also have pdfs. This is already adressed in the documentation section.
- videos to watch and whatnot.
- board size. This is really case by case and doesn't belong in a standard.
- using flux. Yes it makes things easier, no it is not something that needs to be standardized.
- what order you do your assembly and testing in.
While these are useful pieces of information, this is becoming a general knowledge dump and is starting to lose focus.

**Comment by akmorrison** [comment]
The more I read it the more I agree with you, a lot of this advice is out of place. I agree that this document should exist as a thing to check designs/gear against (I should modify the preamble to be clearer on that), and most of this branch doesn't fit into that.

Most of this pull came from me fucking things up in the process of getting PCB's manufactured, and I figured that it would be useful for team members who wanted to do the same, and to spare them the headaches that I dealt with. I know that this all would be better placed in a dev log, but I think it's more general than is served by a buried and project specific note on the drive.

One of the sections in this document when it was first created was titled "General advice from someone who doesn't know what he's doing", and it was intended to hold all of these little things that don't really fit into a standard, but would be useful for new people to read. What are your thoughts on creating a second document just for these little bits of advice? I think it might be helpful for knowledge transfer to new frosh/might result in better designs going forward.

**Comment by akmorrison** [comment]
If we're going to start bringing new people onboard to the electrical department who don't have any experience with electrical stuff (as all people in first year don't), having a resource that will accelerate their knowledge transfer/growth/synergy for team specific stuff would probably be helpful

**Comment by amihaila** [comment]
Yep, I think a general document based on that would be good.

**Comment by akmorrison** [comment]
Aright, I'll remove all but the actual "pass fail" standardy things from this section

**Comment by amihaila** [comment]
A resources folder on the drive might be more convenient to use than a tex doc depending on what we want to include.

**Comment by akmorrison** [comment]
My idea for this is to be printed out and placed on the electronics bench with the standard. Which would make it way more approachable and would increase the frequency by which both we and new recruits read it, which is key.

File: standard.tex [hunk 9]
```diff
@@ -84,20 +84,15 @@ \subsection{Software}
...
-\item BGA components can't be hand-soldered. Bear this in mind while choosing components. There might exist a reflow oven somewhere on campus that we have access to, but laying paste will be a pain. Avoid BGA packages if you plan to assemble by hand.
-\item Don't batch assemble boards (if you need multiple) when they arrive, because you will be less inclined to change your design if you notice small bugs if you've already committed lots of components (which are worth far more than the cheap boards) to this design. Build one, test to death, then decide whether to make another revision or to assemble the rest.
+\item BGA components can't be hand-soldered. There might exist a reflow oven somewhere on campus that we have access to, but laying paste will be a pain. Avoid BGA packages if you plan to assemble by hand.
+\item Don't batch assemble boards (if you are making multiple) when they arrive, because you will be less inclined to change your design if you notice small bugs if you've already committed lots of components (which are worth far more than the cheap boards) to this design. Build one, test to death, then decide whether to make another revision or to assemble the rest.
```
Comment (amihaila) [inline standard.tex]: I'd argue that this doesn't belong.

File: standard.tex (hunk 6 above)
Comment (amihaila) [reply standard.tex]: Maybe rephrase? There isn't really an "I" in this doc anymore, especially if more people add things.

File: standard.tex [hunk 10]
```diff
@@ -84,20 +84,15 @@ \subsection{Software}
...
 \subsection{PCB Design}
 \begin{itemize}
-\item EEVBlog has a great (albeit long) introduction to PCB layout/design: \url{https://www.youtube.com/watch?v=JrH_itjMDjo}. It's like an hour and a half long, but well worth watching.
-\item Given the choice, please draft your schematics and board layouts in KiCad. It's free (GPLv3), used by the rest of the team, and runs on all platforms. \url{http://kicad-pcb.org/}
 \item If your PCBs are meant to be screwed onto something, put mounting holes on your layout, and if at all possible, use M3 screws (for standardization purposes. If you have a space restriction, smaller screws are fine).
-\item The cheapest board that can be purchased in small quantities is a 2 sided 1oz 5cm by 10cm 10/10 (10 mil (.254mm) min clearance and min trace width) 0.4mm smallest drill size board. If at all possible, try to make your design fit that requirement.
 \item If a pin is listed as NC on a data sheet, don't pull it to ground. Just leave it unconnected. Some manufacturers (from what stack overflow tells me), use NC pins for in production testing, and as such they aren't actually high impedance pins, and shorting them elsewhere may destroy your IC.
```
Comment (amihaila) [inline standard.tex]: same as below.

File: standard.tex (hunk 9 above)
Comment (akmorrison) [reply standard.tex]: I'd argue it does. You don't just apply this standard to finished gear, you apply it to designs as well. So if someone makes a layout that has a bga dac on it that they plan to hand solder, that layout design fails.

If you mean the batch assembly bit, that also applies, just because it increases the chances of passing a failed design. This happened with the relay boards, that design is a failure because of the fusing thing, but it also represents $100 of gear that we can't afford to respend, so we're running with it. It may not be a design point, but it's a serious enough consideration to go here.
